vspace/src/main/webapp/WEB-INF/views/staff/files/file.html

```diff
@@ -0,0 +1,124 @@
+<html layout:decorate="~{layouts/main_staff}">
+<head>
+<link rel="stylesheet" type="text/css" id="applicationStylesheet"
+    th:href="@{/resources/extra/Home.css}" />
+
+<script src="//use.fontawesome.com/releases/v5.8.1/js/all.js"
+    data-auto-replace-svg="nest"></script>
+<script th:src="@{'/resources/extra/space.js'}"></script>
+
+<script th:inline="javascript">
+
+function base64ToArrayBuffer(base64) {
+    var binaryString = window.atob(base64);
+    var binaryLen = binaryString.length;
+    var bytes = new Uint8Array(binaryLen);
+    for (var i = 0; i < binaryLen; i++) {
+       var ascii = binaryString.charCodeAt(i);
+       bytes[i] = ascii;
+    }
+    return bytes;
+ }
+ 
+function saveByteArray(reportName, byte) {
+	var fileType = [[${file.fileType}]];
+    var blob = new Blob([byte], {type: fileType});
+    var link = document.createElement('a');
+    link.href = window.URL.createObjectURL(blob);
+    var fileName = reportName;
+    link.download = fileName;
+    link.click();
+};
+
+function downloadFile(){
+	$.ajax({
+        url: "[(@{'/staff/files/download/'+${file.id}})]",
+        type: 'GET',
+        cache       : false,
+        contentType : false,
+        success: function(data) {
+        	var sampleArr = base64ToArrayBuffer(data);
+        	var fileName = [[${file.filename}]];
+        	console.log(fileName)
+        	saveByteArray(fileName, sampleArr);
+        },
+        error: function(data) {
+        	console.log('could not download')
+        }
+    });
+}
+
+function deleteFile(){
+	var token = [[${_csrf.token}]];
+	$.ajax({
+        url: "[(@{'/staff/files/delete/'+${file.id}})]",
+        type: 'POST',
+        headers:{
+			'X-CSRF-Token':token
+		},
+        cache: false,
+        contentType : false,
+        success: function(data) {
+        	window.location.href = "[(@{'/staff/files/list/'})]";
+        },
+        error: function(data) {
+        	var alert = $('<div class="alert alert-danger alert-dismissible fade show" role="alert"><p>We are sorry but something went wrong. Please try again later.</p><button type="button" class="close" data-dismiss="alert" aria-label="Close"><span aria-hidden="true">&times;</span></button></div>');
+            $('.errorDelete').append(alert);
+        }
+    });
+}
+
+</script>
+<style>
+/* .download{
+margin-left:1200px;
+} */
+</style>
+</head>
+
+<body>
+    <div  layout:fragment="content" class="main-content">
+    <span class = "errorDelete"></span>
+ 		<div class="flex-spacing">
+ 			<div>
+	    		<h2 th:text=${file.filename}/>
+		        <p>[[${file.fileDescription}]]</p>
+		        
+		        
+	    	</div>
+ 			 
+ 		</div>
+ 		<div>
+ 		<button class="download btn primary-btn" type="submit" value="submit" onClick = "deleteFile()" style="float:right">
```

Comment:
why not just linking to the file download url instead of using javascript?

Reply:
Didn't get you, javascript is processing the response in success scenario for both download and delete.

Reply:
ah, now I get it. I just saw the css class called "download" and assumed it was the download link. Should. probably be renamed ;)